src/main/kotlin/ink/abb/pogo/scraper/tasks/Walk.kt

```diff
@@ -77,22 +78,24 @@ class Walk(val sortedPokestops: List<Pokestop>, val lootTimeouts: Map<String, Lo
 
         var walking = true
         bot.runLoop(timeout, "WalkingLoop") { cancel ->
-            // don't run away when there are still Pokemon around
-            if(walking) {
-                if(bot.api.map.getCatchablePokemon(ctx.blacklistedEncounters).size > 0 && settings.shouldCatchPokemons) {
-                    // Stop walking
-                    walking = false
-                    Log.normal("Pausing to catch pokemon...")
-                } // Else continue walking.
-            } else {
-                if(bot.api.map.getCatchablePokemon(ctx.blacklistedEncounters).size <= 0) {
-                    walking = true
-                    Log.normal("Resuming walk.")
-                } // Else continue waiting.
-            }
+            if (ctx.api.inventories.itemBag.hasPokeballs()) {
```

Comment:
This line should be further below - in the `if` statement that checks if walking should pause.  Otherwise it could cause the bot not to start walking again when you use your last pokemon.